payjoin/src/send/v2/error.rs

```diff
@@ -120,3 +122,23 @@ impl From<InternalEncapsulationError> for super::ResponseError {
         )
     }
 }
+
+impl From<DirectoryResponseError> for EncapsulationError {
+    fn from(value: DirectoryResponseError) -> Self {
+        match value {
+            DirectoryResponseError::InvalidSize(e) => InternalEncapsulationError::InvalidSize(e),
+            DirectoryResponseError::OhttpDecapsulation(e) => InternalEncapsulationError::Ohttp(e),
+            DirectoryResponseError::UnexpectedStatusCode(e) =>
+                InternalEncapsulationError::UnexpectedStatusCode(e),
+        }
+        .into()
+    }
+}
+
+impl From<DirectoryResponseError> for ResponseError {
+    fn from(value: DirectoryResponseError) -> Self {
+        ResponseError::Validation(
+            super::InternalValidationError::V2Encapsulation(value.into()).into(),
+        )
+    }
+}
```

Comment:
Same as above, it seems like this conversion is only done once (which could be done with map_err or an Err(e) match arm) making the From unnecessarily verbose, and the EncapsulationError seems like it could just have a `DirectoryResponseError` unless these `InvalidSize`, `OhttpDecapsulation`, `UnexpectedStatusCode` variants crop up outside of the context of a directory response.